dev-guidelines/common/DEPRECATION-GUIDELINES.md

```diff
@@ -0,0 +1,30 @@
+# DEPRECATION GUIDELINES
+
+Every feature removal should be given careful thought. Gently deprecating things (keep supporting them for 1+ release with a console warning) has a big value and is the preference.
+
+* But sometimes the price in e.g. complexity or code overhead can be too big (thinking about deck.gl v4 which has three big deprecated ChoroplethLayers that take up a lot of space in the app bundle but are used by very few apps. One could question whether keeping them was the right choice). The situation is similar for the less frequently used Overlays in react-map-gl v3.
+
+* In cases where we can't or won't do gentle deprecation, we should document the breaking changes in an upgrade guide.
+
+* I'm in favor of following what Facebook does with React in general - console.warn for API deprecations and clearly stating in the CHANGELOG and then deprecate 2-3 versions later.
+
+* The question also is when to eventually do the deprecation. Do we do it in a minor patch, or wait for a major version bump?
+
+ * iven we already have a few majors, everything that is potentially breaking should be done in a major
+ @ibgreen
+
+* Isn't the key consideration that we should provide some reasonable time for users to update? If a feature has been deprecated for 6+ months, and we are cutting a new major release, it would seem acceptable to remove it.
```

Comment:
This paragraph sounds like a note... we should rephrase the first sentence (the question) as a statement.